### PR TITLE
Add signTransaction

### PR DESCRIPTION
**Pull request by luizstacio**
# Summary
- Create a new package `@fuels-ts/testcases` with mock transactions data. Also, move the currently used ones to these pack.
- Update to use `canonical signature` on `ellipict.sign`.
- Create `Hasher.hashTransaction` it create an `txid` as specified on [here](FuelLabs/fuel-specs/blob/master/specs/protocol/tx_validity.md#valid-signatures).
- Create `Wallet.signTransaction`.

closes: #39

### CODE REVIEW

**Review by QuinnLee — APPROVED** [review]

File: packages/wallet/src/hasher.ts [hunk 1]
```diff
@@ -11,6 +16,51 @@ export function hashMessage(msg: string) {
...
+  }
+
+  // Zero out input fields
+  transaction.inputs = transaction.inputs.map((input) => {
+    const inputClone = { ...input };
+
+    switch (inputClone.type) {
+      // Zero out on signing: txoPointer
+      case InputType.Coin: {
+        // inputClone.txoPointer = 0;
```
Comment (digorithm) [inline packages/wallet/src/hasher.ts:40]: If I understand correctly if the transaction's input is a coin, shouldn't we set the `txoPointer` to `0` when signing? Why is this commented out?

File: packages/wallet/src/hasher.ts [hunk 2]
```diff
@@ -11,6 +16,51 @@ export function hashMessage(msg: string) {
...
+        return inputClone;
+      }
+      // Zero out on signing: txID, outputIndex, balanceRoot, stateRoot, and txoPointer
+      case InputType.Contract: {
+        // inputClone.txoPointer;
```
Comment (digorithm) [inline packages/wallet/src/hasher.ts:45]: Same here, shouldn't we set `txoPointer` to `0`?

File: packages/wallet/src/hasher.ts (hunk 1 above)
Comment (luizstacio) [reply packages/wallet/src/hasher.ts:40]: We didn't have the field `txoPointer` yet on the transaction.

Comment (digorithm) [reply packages/wallet/src/hasher.ts:40]: Oh I see, makes sense then.

**Pull request merged**